experimental/saiexperimentaldashha.h

```diff
@@ -0,0 +1,442 @@
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiexperimentaldashha.h
+ *
+ * @brief   This module defines SAI P4 extension  interface
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHHA_H_)
+#define __SAIEXPERIMENTALDASHHA_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASH_HA SAI - Extension specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute ID for DASH HA
+ */
+typedef enum _sai_dash_ha_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_DASH_HA_ATTR_START,
+
+    /**
+     * @brief HA Local IP for control traffic
+     *
+     * @type sai_ip_address_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     */
+    SAI_DASH_HA_ATTR_CONTROL_LOCAL_IP = SAI_DASH_HA_ATTR_START,
+
+    /**
+     * @brief HA Peer IP for control traffic
+     *
+     * @type sai_ip_address_t
+     * @flags CREATE_AND_SET
+     * @default 0.0.0.0
+     */
+    SAI_DASH_HA_ATTR_CONTROL_PEER_IP,
+
+    /**
+     * @brief List of HA VIPs
+     *
+     * @type sai_dash_ha_vip_info_list_t
+     * @flags CREATE_AND_SET
+     * @default empty
+     */
+    SAI_DASH_HA_ATTR_VIP_INFO_LIST,
+
+    /**
+     * @brief HA heartbeat interval time in ms
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_DASH_HA_ATTR_HEARTBEAT_INTERVAL,
+
+    /**
+     * @brief Number of HA heartbeats missed before declaring Peer unreachable
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_DASH_HA_ATTR_HEARTBEAT_COUNT,
+
+    /**
+     * @brief Time in ms required for network to converge after switchover
+     *
+     * @par Any traffic directed to wrong DPU after this time will be blackholed
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 100
+     */
+    SAI_DASH_HA_ATTR_VIP_CONVERGENCE_TIMEOUT,
+
+    /**
+     * @brief List of HA VIP status
+     *
+     * @type sai_dash_ha_vip_status_list_t
+     * @flags READ_ONLY
+     */
+    SAI_DASH_HA_ATTR_VIP_STATUS_LIST,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_DASH_HA_ATTR_END,
+
+    /** Custom range base value */
+    SAI_DASH_HA_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_DASH_HA_ATTR_CUSTOM_RANGE_END
+
+} sai_dash_ha_attr_t;
```

Comment:
this is essentially the HA set, so we could align using the HA set concept.

Reply:
One mismatch is that here the HA object has multiple VIP ID support for the same DPU pair.